### ISSUE TITLE
FirebaseMessaging used API types not declared in its Privacy Manifest

### ISSUE THREAD

**Issue by aureliencolas** [firebase/firebase-ios-sdk#12561]
### Description

When uploading the iOS app to AppStoreConnect, I get the "ITMS-91053: Missing API declaration" warnings for the NotificationService app extension (embedded in the app for push notifications support) and for the following API categories:
- NSPrivacyAccessedAPICategoryUserDefaults
- NSPrivacyAccessedAPICategoryDiskSpace
- NSPrivacyAccessedAPICategoryFileTimestamp

The NotificationService class only includes FirebaseMessaging and does not use UserDefaults nor disk space or file timestamp APIs itself. Here is the whole code (more or less the sample code provided by Firebase):

    import UserNotifications
    import FirebaseMessaging

    class NotificationService: UNNotificationServiceExtension {
        var contentHandler: ((UNNotificationContent) -> Void)?
        var bestAttemptContent: UNMutableNotificationContent?

        override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
            self.contentHandler = contentHandler
            if let content = (request.content.mutableCopy() as? UNMutableNotificationContent) {
                bestAttemptContent = content
                FIRMessagingExtensionHelper().populateNotificationContent(content, withContentHandler: contentHandler)
            }
        }
        
        override func serviceExtensionTimeWillExpire() {
            guard let bestAttemptContent = bestAttemptContent else { return }
            contentHandler?(bestAttemptContent)
        }
    }

Since Firebase iOS SDK 10.22.1, the different Firebase modules now expose a Privacy manifest but there is nothing in the FirebaseMessaging privacy manifest regarding the use of these APIs.

Additionally, since the app also includes FirebaseMessaging, I also get these warnings for the app itself. Besides, the app also embeds FirebaseCrashlytics and FirebaseAnalyticsWithoutAdIdSupport modules which may also use these APIs (the main app does not either) but it's another topic.

### Reproducing the issue

To reproduce the issue, add to an iOS app a NotificationService app extension which includes FirebaseMessaging and upload the app to AppStoreConnect.

### Firebase SDK Version

10.22.1

### Xcode Version

15.2

### Installation Method

Swift Package Manager

### Firebase Product(s)

Messaging

### Targeted Platforms

iOS, tvOS, App Extensions

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

{
  "pins" : [
    {
      "identity" : "abseil-cpp-binary",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/abseil-cpp-binary.git",
      "state" : {
        "revision" : "df308b8b46607675f2b9ec8e569109008f9155ce",
        "version" : "1.2022062300.1"
      }
    },
    {
      "identity" : "app-check",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/app-check.git",
      "state" : {
        "revision" : "3e464dad87dad2d29bb29a97836789bf0f8f67d2",
        "version" : "10.18.1"
      }
    },
    {
      "identity" : "appauth-ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/openid/AppAuth-iOS.git",
      "state" : {
        "revision" : "7e2c09cbeb3bb799f26c268dbedb26325ea722a9",
        "version" : "1.7.3"
      }
    },
    {
      "identity" : "chargebee-ios-tvos",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/aureliencolas/chargebee-ios-tvos",
      "state" : {
        "branch" : "master",
        "revision" : "a53c78e9d3d5e57004fea99e60312bdd2ae1c3fa"
      }
    },
    {
      "identity" : "facebook-ios-sdk",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/facebook/facebook-ios-sdk",
      "state" : {
        "revision" : "e14f349945ba3f9516e3292107f42be19fe9fbe5",
        "version" : "12.3.2"
      }
    },
    {
      "identity" : "firebase-ios-sdk",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/firebase-ios-sdk.git",
      "state" : {
        "revision" : "be49849dcba96f2b5ee550d4eceb2c0fa27dade4",
        "version" : "10.22.1"
      }
    },
    {
      "identity" : "googleappmeasurement",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleAppMeasurement.git",
      "state" : {
        "revision" : "482cfa4e5880f0a29f66ecfd60c5a62af28bd1f0",
        "version" : "10.22.1"
      }
    },
    {
      "identity" : "googledatatransport",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleDataTransport.git",
      "state" : {
        "revision" : "a637d318ae7ae246b02d7305121275bc75ed5565",
        "version" : "9.4.0"
      }
    },
    {
      "identity" : "googlesignin-ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleSignIn-iOS.git",
      "state" : {
        "revision" : "7932d33686c1dc4d7df7a919aae47361d1cdfda4",
        "version" : "7.0.0"
      }
    },
    {
      "identity" : "googleutilities",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleUtilities.git",
      "state" : {
        "revision" : "26c898aed8bed13b8a63057ee26500abbbcb8d55",
        "version" : "7.13.1"
      }
    },
    {
      "identity" : "grpc-binary",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/grpc-binary.git",
      "state" : {
        "revision" : "ea4cb5cc0c39c732b85386263116d2e2fdbbdc61",
        "version" : "1.49.2"
      }
    },
    {
      "identity" : "gtm-session-fetcher",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/gtm-session-fetcher.git",
      "state" : {
        "revision" : "76135c9f4e1ac85459d5fec61b6f76ac47ab3a4c",
        "version" : "3.3.1"
      }
    },
    {
      "identity" : "gtmappauth",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GTMAppAuth.git",
      "state" : {
        "revision" : "cee3c709307912d040bd1e06ca919875a92339c6",
        "version" : "2.0.0"
      }
    },
    {
      "identity" : "interop-ios-for-google-sdks",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/interop-ios-for-google-sdks.git",
      "state" : {
        "revision" : "2d12673670417654f08f5f90fdd62926dc3a2648",
        "version" : "100.0.0"
      }
    },
    {
      "identity" : "kingfisher",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/onevcat/Kingfisher.git",
      "state" : {
        "revision" : "5b92f029fab2cce44386d28588098b5be0824ef5",
        "version" : "7.11.0"
      }
    },
    {
      "identity" : "leveldb",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/leveldb.git",
      "state" : {
        "revision" : "43aaef65e0c665daadf848761d560e446d350d3d",
        "version" : "1.22.4"
      }
    },
    {
      "identity" : "lottie-spm",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/airbnb/lottie-spm.git",
      "state" : {
        "revision" : "3bd43e12d6fb54654366a61f7cfaca787318b8ce",
        "version" : "4.4.1"
      }
    },
    {
      "identity" : "nanopb",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/nanopb.git",
      "state" : {
        "revision" : "b7e1104502eca3a213b46303391ca4d3bc8ddec1",
        "version" : "2.30910.0"
      }
    },
    {
      "identity" : "promises",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/promises.git",
      "state" : {
        "revision" : "540318ecedd63d883069ae7f1ed811a2df00b6ac",
        "version" : "2.4.0"
      }
    },
    {
      "identity" : "sentry-cocoa",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/getsentry/sentry-cocoa.git",
      "state" : {
        "revision" : "cf43eac1aa12017868c257ad3854ad87a5de0758",
        "version" : "7.31.5"
      }
    },
    {
      "identity" : "swift-protobuf",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/apple/swift-protobuf.git",
      "state" : {
        "revision" : "65e8f29b2d63c4e38e736b25c27b83e012159be8",
        "version" : "1.25.2"
      }
    },
    {
      "identity" : "swiftkeychainwrapper",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/jrendel/SwiftKeychainWrapper",
      "state" : {
        "revision" : "185a3165346a03767101c4f62e9a545a0fe0530f",
        "version" : "4.0.1"
      }
    },
    {
      "identity" : "swiftyxml",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/chenyunguiMilook/SwiftyXML.git",
      "state" : {
        "revision" : "30040f38675bc3a1666a0b0f6417adabaf26a3c6",
        "version" : "3.1.0"
      }
    }
  ],
  "version" : 2
}


```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by r3econ** [comment]
The problem affects also our latest app submission.

**Comment by ncooke3** [comment]
Hi all, let's continue the conversation in https://github.com/firebase/firebase-ios-sdk/issues/12557.

It seems that the validation tooling is not picking up the SDK-specific privacy manifests when libraries are being statically linked. We have filed feedback tickets with Apple for further guidance. I'll keep #12557 updated as we learn more.

**Comment by saikumar-osv** [comment]
Still getting same warning Apple.

**Comment by binouze** [comment]
Hi, I see the issue is closed, is the fix available in the release 10.23.0 ?

**Comment by paulb777** [comment]
See the comment [above](https://github.com/firebase/firebase-ios-sdk/issues/12561#issuecomment-2000599939). Now tracked at #12557
